build_tools/github_actions/lint_check_license.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+# Copyright 2024 The StableHLO Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+print_usage() {
+  echo "Usage: $0 [-b]"
+  echo "    -b <branch>  Base branch name, defaults to main."
+}
+
+BASE_BRANCH=main
+while getopts 'b:' flag; do
+  case "${flag}" in
+    b) BASE_BRANCH="$OPTARG" ;;
+    *) print_usage
+       exit 1 ;;
+  esac
+done
+shift $(( OPTIND - 1 ))
+
+if [[ $# -ne 0 ]] ; then
+  print_usage
+  exit 1
+fi
+
+echo "Gathering changed files..."
+echo
+mapfile -t CHANGED_FILES < <(git diff "$BASE_BRANCH" HEAD --name-only --diff-filter=d)
+if (( ${#CHANGED_FILES[@]} == 0 )); then
+  echo "Found no changed files."
+  exit 0
+fi
+
+echo "Checking the following files for licenses:
+$(printf "%s\n" "${CHANGED_FILES[@]}")"
+echo
+
+UNLICENSED_FILES=()
+for file in "${CHANGED_FILES[@]}"; do
+  if [[ "$file" == MODULE.bazel.lock ]]; then
+    echo "Skipping generated file: $file"
+    echo
+    continue
+  fi
+  if [[ "$file" = *.mlir ]]; then
+    echo "Skipping MLIR assembly file: $file"
+    echo
+    continue
+  fi
+  if [[ "$file" = *.md ]]; then
+    echo "Skipping Markdown file: $file"
+    echo
+    continue
+  fi
+  if ! head -20 "$file" | grep "Copyright" &>/dev/null; then
```

Comment:
Thanks for the feedback! My thinking with this PR was to retire `addlicense` in favor of a simpler mechanism. The current PR matches the functionality provided by `addlicense` (in fact, it even exceeds it because `addlicense` does not support all the kinds of files we care about, but that is a side effect of using `find` and did not require any extra effort).

I agree that it would be best for this check to be comprehensive so that we detect issues as soon as possible. Would it be okay with you if we expand the check in a follow-up PR?

Reply:
Makes sense. 
Sure.